From 555bdd908e31e9ef5458f3f19cf6143e622a8769 Mon Sep 17 00:00:00 2001
From: "Raj, Deepu" <deepu.raj@ti.com>
Date: Mon, 21 Nov 2011 12:29:20 +0530
Subject: [PATCH 129/193] ARM: OMAP2+: ALSA : ASoC tlv320aic3x.c
 suspend/resume cache_sync fix

 This patch fixes audio resume failure. AIC3X codec driver is not
 generating disable regulator event due to unknown reason.
 Need to revisit the patch to find the reason and fix the same.

Signed-off-by: Raj, Deepu <deepu.raj@ti.com>
Signed-off-by: Hebbar, Gururaja <gururaja.hebbar@ti.com>
---
 sound/soc/codecs/tlv320aic3x.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sound/soc/codecs/tlv320aic3x.c b/sound/soc/codecs/tlv320aic3x.c
index 87d5ef1..d0dbac1 100644
--- a/sound/soc/codecs/tlv320aic3x.c
+++ b/sound/soc/codecs/tlv320aic3x.c
@@ -1147,6 +1147,11 @@ static int aic3x_set_power(struct snd_soc_codec *codec, int power)
 		codec->cache_only = 1;
 		ret = regulator_bulk_disable(ARRAY_SIZE(aic3x->supplies),
 					     aic3x->supplies);
+		/* Enable cache sync if regulator disable
+		 * event is not triggerd.
+		 * ToDo : Revisit later to fix it
+		 */
+		codec->cache_sync = 1;
 	}
 out:
 	return ret;
-- 
1.7.9.5

